### ISSUE TITLE
Failed to build test\core\surface\byte_buffer_reader_test.c on Windows

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#4050]
https://grpc-testing.appspot.com/job/gRPC_master/12733/config=dbg,language=c,platform=windows/consoleFull

I think the relevant part is:
```
..\..\..\..\test\core\surface\byte_buffer_reader_test.c(197): error C2220: warning treated as error - no 'object' file generated [c:\jenkins\workspace\gRPC_master@2\config\dbg\language\c\platform\windows\vsprojects\vcxproj\test\grpc_byte_buffer_reader_test\grpc_byte_buffer_reader_test.vcxproj]
```

**Comment by jtattermusch** [comment]
This is preventing build, so marking as P0

**Comment by jtattermusch** [comment]
Exactly the same issue also for opt  build on Windows:
https://grpc-testing.appspot.com/job/gRPC_master/12733/config=opt,language=c,platform=windows/console

**Comment by ctiller** [comment]
706>..\..\..\..\test\core\surface\byte_buffer_reader_test.c(197): warning
C4090: 'function' : different 'const' qualifiers
[c:\jenkins\workspace\gRPC_master@2\config\dbg\language\c\platform\windows\vsprojects\vcxproj\test\grpc_byte_buffer_reader_test\grpc_byte_buffer_reader_test.vcxproj]
706>..\..\..\..\test\core\surface\byte_buffer_reader_test.c(198): warning
C4090: 'function' : different 'const' qualifiers
[c:\jenkins\workspace\gRPC_master@2
\config\dbg\language\c\platform\windows\vsprojects\vcxproj\test\grpc_byte_buffer_reader_test\grpc_byte_buffer_reader_test.vcxproj]

**Comment by ctiller** [comment]
I think just drop the const from the declarations.
Msvc is wrong in this respect, but what can we do?

**Comment by dgquintas** [comment]
On it. Will send fix PR post-dinner.

**Comment by dgquintas** [comment]
#4053. Waiting for green now.

**Comment by dgquintas** [comment]
Fixed by #4053
